common/js_handle.go

```diff
@@ -189,12 +172,17 @@ func (h *BaseJSHandle) JSONValue() string { // TODO: return error
 			WithAwaitPromise(true).
 			WithObjectID(h.remoteObject.ObjectID)
 		if result, _, err = action.Do(cdp.WithExecutor(h.ctx, h.session)); err != nil {
-			k6ext.Panic(h.ctx, "getting properties for JS handle: %w", err)
+			return "", fmt.Errorf("retrieving json value: %w", err)
 		}
 		remoteObject = result
 	}
 
-	return convertToString(remoteObject)
+	res, err := parseConsoleRemoteObject(h.logger, remoteObject)
+	if err != nil {
+		return "", fmt.Errorf("extracting json value: %w", err)
```

Comment:
The comment [here](https://github.com/grafana/xk6-browser/pull/1190#discussion_r1469291804) explains:
> I guess the two error messages made sense for debug purposes to help clarify which conditional branch the method was working on.

What about adding the object ID in the error so that we can see which path is taken?

```suggestion
		return "", fmt.Errorf("extracting json value (remote object id: %v): %w", remoteObject.ObjectID, err)
```